### PR TITLE
Oss/new pr

### PR DESCRIPTION
**Pull request by OsPit01**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import java.util.Scanner;
 
 public class Main {
+
     private final static Scanner scanner = new Scanner(System.in);
```
Comment (VladOsipov) [inline src/Main.java]: private static final

File: src/PrintUsersCommand.java [hunk 2]
```diff
@@ -1,11 +1,5 @@
-import java.util.Set;
-
 public class PrintUsersCommand {
     public void printUsers() {
-
-        Set<String> userNames = UserContainer.getUsers().keySet();
-        for (String userName : userNames) {
-            System.out.print("username" + " " + userName  + "\n");
-        }
+        System.out.println(UserContainer.getUsers().toString());
```
Comment (VladOsipov) [inline src/PrintUsersCommand.java]: sout calls `toString` method inside automatically

File: src/User.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public String toString() {
+        return "User:" + " " + username  + " " +
+                "password:" + " " + password + " " +
+                "name:" + " " + name + " " +
+                "surname:" + " " + surname + "\n";
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (VladOsipov) [inline src/User.java]: where do you use this method?

File: src/UserContainer.java [hunk 4]
```diff
@@ -1,19 +1,23 @@
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 
 public class UserContainer {
-   private final static Map<String, String> users = new HashMap<String, String>();
 
-    public static Map<String, String> getUsers() {
+    private final static List<User> users = new ArrayList();
```
Comment (VladOsipov) [inline src/UserContainer.java]: private static final

File: src/UserContainer.java [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
 
     public static boolean isUserNameAndPasswordExists(String userName, String password) {
-        boolean userExists = false;
-
-        if (users.containsKey(userName) && users.containsValue(password)) {
-            userExists = true;
+        boolean existUser = false;
+        for (int i = 0; i < users.size(); i++) {
+            if (users.get(i).getUsername().equals(userName)) {
+                if (users.get(i).getPassword().equals(password)) {
```
Comment (VladOsipov) [inline src/UserContainer.java]: maybe it is easier to find user
and then get username and get passowrd from this user?

now you get user by index every time

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/LoginCommand.java [hunk 6]
```diff
@@ -1,7 +1,8 @@
 
 public class LoginCommand {
-    public void execute(String userName, String userPassword) {
-        if (UserContainer.isUserNameAndPasswordExists(userName, userPassword)) {
+    public void execute(String userName, String Password) throws UserNotFoundException {
```
Comment (VladOsipov) [inline src/LoginCommand.java]: Why did you change `userPassword` param name to `Password`
it is ok to name it `password`, but do not use capital letters for variables

Comment (VladOsipov) [reply src/LoginCommand.java]: Do you need throws here?

File: src/Main.java [hunk 7]
```diff
@@ -1,9 +1,10 @@
 import java.util.Scanner;
 
 public class Main {
-    private final static Scanner scanner = new Scanner(System.in);
 
-    public static void main(String[] args) {
+    private  static final Scanner scanner = new Scanner(System.in);
```
Comment (VladOsipov) [inline src/Main.java]: ctrl + alt + L

File: src/Main.java [hunk 8]
```diff
@@ -1,9 +1,10 @@
...
+
+    public static void main(String[] args) throws UserNotFoundException {
```
Comment (VladOsipov) [inline src/Main.java]: Do you need `throws` here ?

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/LoginCommand.java (hunk 6 above)
Comment (OsPit01) [reply src/LoginCommand.java]: it is an accident

File: src/UserContainer.java [hunk 9]
```diff
@@ -1,19 +1,27 @@
...
-
-        if (users.containsKey(userName) && users.containsValue(password)) {
-            userExists = true;
+    public static boolean isUserNameAndPasswordExists(String userName, String password) throws UserNotFoundException {
+        boolean existUser = false;
+        for (int i = 0; i < users.size(); i++) {
+            User foundUser = users.get(i);
+            if (foundUser.getUsername().equals(userName)) {
+                if (foundUser.getPassword().equals(password)) {
+                    existUser = true;
```
Comment (VladOsipov) [inline src/UserContainer.java]: Do you need to go further through your users list in case you've found user?
you will iterate over ALL the list

**Review by VladOsipov — CHANGES_REQUESTED** [review]
